systemd-swap

```diff
@@ -318,6 +318,29 @@ case "$1" in
     fi
 
     if YN "${swapfc_enabled}"; then
+      # validate swapfc_frequency due to possible issues caused if set incorrectly
+      if [[ -n "${swapfc_frequency//[0-9]}" || -z "$swapfc_frequency" ]]; then
+        ERRO "swapfc_frequency is not set to a valid integer value: ${swapfc_frequency}"
+      elif (( swapfc_frequency < 1 || swapfc_frequency > 86400 )); then
+        WARN "swapfc_frequency must be in range of 1..86400: ${swapfc_frequency} - set to 1"
+        swapfc_frequency=1
+      fi
+
+      polling_rate=$swapfc_frequency
+      double_polling_rate() {
+        local val=$(( polling_rate * 2 ))
+        # do not double, interval is long enough
+        (( val > 86400 || val > swapfc_frequency * 1000 )) && return
+        polling_rate=$val
+        WARN "swapFC: polling rate doubled to ${polling_rate}s"
+      }
+      reset_polling_rate() {
+        if (( polling_rate > swapfc_frequency )); then
+          polling_rate=$swapfc_frequency
+          INFO "swapFC: polling rate reset to ${polling_rate}s"
+        fi
+      }
```

Comment:
The check for `polling_rate > swapfc_frequency` is for preventing INFO message logging, since `reset_polling_rate` is called withing `create_swapfile()` no matter what.

Reply:
Ahh, but wouldn't it semantically make more sense to have the if statement around the call for `reset_polling_rate` instead of inside it?

Reply:
That's up to you. I did it this way to eventually reuse `reset_polling_rate` and not have to write that logging statement again.

Reply:
If we reuse the function you're absolutely correct